src/components/CreatePost/SubTravelogue.tsx

```diff
@@ -0,0 +1,164 @@
+import { AddCircle as AddCircleIcon } from '@mui/icons-material';
+import { Box, Button, IconButton, OutlinedInput, Stack } from '@mui/material';
+import { styled } from '@mui/material/styles';
+import { useEffect } from 'react';
+import { Controller, useFieldArray, useForm } from 'react-hook-form';
+import useFormPersist from 'react-hook-form-persist';
+
+import { usePostSubTravelogue } from '@/api/hooks/post';
+import { SubTitle } from '@/components/common';
+import { subTravelogueFormDefault } from '@/constants/defaultFormValue';
+import useSubTravelogueForm from '@/hooks/useSubTravelogueForm';
+import { ButtonEventType } from '@/types/common';
+import { StepType, SubTravelogueType } from '@/types/post';
+
+import { Location, Transportation } from '.';
+
+interface SubTravelogueProps {
+  travelogueId: string;
+  isLastStep: boolean;
+  index: number;
+  handleStep: (e: ButtonEventType, type: StepType) => void;
+  handleComplete: (isValid?: boolean) => void;
+}
+
+const SubTravelogue = ({
+  travelogueId,
+  isLastStep,
+  index,
+  handleStep,
+  handleComplete,
+}: SubTravelogueProps) => {
+  const saveData = JSON.parse(
+    sessionStorage.getItem(`save-${travelogueId}-${index}`) as string,
+  );
+  const {
+    control,
+    handleSubmit,
+    watch,
+    setValue,
+    reset,
+    formState: { isDirty },
+  } = useForm<SubTravelogueType>({
+    defaultValues: saveData ?? subTravelogueFormDefault,
+  });
+  const { title, content, transportationSet } = useSubTravelogueForm(control);
+  const { fields, append, remove } = useFieldArray({ control, name: 'addresses' });
+  const { mutate } = usePostSubTravelogue();
+  useFormPersist(`temp-${travelogueId}-${index}`, {
+    watch,
+    setValue,
+  });
+
+  const fieldValue = watch();
+  const isFormEmpty =
+    JSON.stringify(fieldValue) === JSON.stringify(subTravelogueFormDefault);
+
+  useEffect(() => {
+    if (!isFormEmpty) {
+      handleComplete(isDirty);
+    }
+  }, [isDirty]);
```

Comment:
ask;

```suggestion
  }, [handleComplete, isDirty, isFormEmpty]);
```

isFormEmpty, handleComplete 상태를 의존성이 넣지 않은 이유가 있나요?

Reply:
handleComplete를 넣으면 불필요하게 실행이 발생하고, 로직에서 isFormEmpty의 값이 바뀐다고 실행하는 것을 원치 않았기에 넣지 않았습니다.